lib/src/interceptor.dart

```diff
@@ -68,7 +76,30 @@ class Interceptor {
 
   StreamController? _onReply;
 
-  Interceptor(this._matcher);
+  Interceptor(this._matcher, {this.generateAuto = true});
+
+  Interceptor? get autoOption {
+    if (!generateAuto) return null;
+    switch (_matcher.method.toUpperCase()) {
+      case 'GET' : return copyWith(
```

Comment:
Similar comment to (now-outdated): I apologise for this. I think the correct way to handle this is to create a separate subclass for each method type (and throwing clause), and put this kind of behaviour in there. But that seemed like a fairly major piece of refactoring, and I didn't want to do that without your approval. But, I wanted you to know, I wouldn't have done this in my code :-)